vllm/model_executor/layers/fused_moe/layer.py

```diff
@@ -83,6 +83,15 @@ def create_weights(self, layer: torch.nn.Module, num_experts: int,
         layer.register_parameter("w2_weight", w2_weight)
         set_weight_attrs(w2_weight, extra_weight_attrs)
 
+    def process_weights_after_loading(self, layer: torch.nn.Module) -> None:
+        super().process_weights_after_loading(layer)
+
+        if current_platform.is_cpu():
```

Comment:
Since I assume IPEX only will work for x86, could you include a check for cpu architecture i.e. 
```
if current_platform.is_cpu() and current_platform.get_cpu_architecture() is CpuArchEnum.X86:
```

If this is the case, then we should move the NotImplementedError here where we can check for CPU platforms that aren't supported by IPEX

Reply:
Good point, updated.